cpp/src/io/parquet/page_decode.cuh

```diff
@@ -901,17 +901,10 @@ inline __device__ uint32_t InitLevelSection(page_state_s* s,
   // this is a little redundant. if level_bits == 0, then nothing should be encoded
   // for the level, but some V2 files in the wild violate this and encode the data anyway.
   // thus we will handle V2 headers separately.
-  if ((s->page.flags & PAGEINFO_FLAGS_V2) != 0) {
+  if ((s->page.flags & PAGEINFO_FLAGS_V2) != 0 && (len = s->page.lvl_bytes[lvl]) != 0) {
     // V2 only uses RLE encoding so no need to check encoding
-    len = lvl == level_type::DEFINITION ? s->page.def_lvl_bytes : s->page.rep_lvl_bytes;
     s->abs_lvl_start[lvl] = cur;
-    if (len == 0) {
-      s->initial_rle_run[lvl]   = s->page.num_input_values * 2;  // repeated value
-      s->initial_rle_value[lvl] = 0;
-      s->lvl_start[lvl]         = cur;
-    } else {
-      init_rle(cur, cur + len);
-    }
+    init_rle(cur, cur + len);
```

Comment:
Why do we have such big change here? Will this a breaking change?

Reply:
it's because now I set and test `len` in the `if`, so `len` is guaranteed non-zero and I can get rid of the redundant branching for the `len==0` case.

Reply:
Oh, do we now hit the else branch at 915 instead? I see similar code with the branch that's removed here.

Reply:
Yes, exactly.  The original fix from #13707 created a duplicate path to get to 0 bits for the level RLE encoding. This refactor gets rid of the duplicate.